frame/transaction-payment/src/lib.rs

```diff
@@ -207,14 +205,18 @@ where
 		let normal_block_weight =
 			current_block_weight.get(DispatchClass::Normal).min(normal_max_weight);
 
-		// TODO: Handle all weight dimensions
-		let normal_max_weight = normal_max_weight.ref_time();
-		let normal_block_weight = normal_block_weight.ref_time();
+		// take only the scarcer resource into account.
+		let (normal_block_weight, normal_max_weight) =
+			if normal_block_weight.ref_time() > normal_block_weight.proof_size() {
```

Comment:
You can *never* compare `proof_size` with `ref_time` directly.  
I did something different in the other MR; first converting them to a currency and them comparing the currency values.

I think the correct approach here is to calculate the ratio per dimension to the max block and select by the larger ration.  
So for example a call with weight `[2, 3]` and a max-block weight of `[3, 6]` would be limited by the first dimension since `2/3 > 3/6`.